src/android/Device.java

```diff
@@ -59,31 +69,118 @@ public void initialize(CordovaInterface cordova, CordovaWebView webView) {
     }
 
     /**
-     * Executes the request and returns PluginResult.
+     * Executes the request and stores all collected
+     * device info (like OS, platform, model, serial etc...)
+     * in a JSONObject (= PluginResult).
+     *
+     * In case of a zebra device running Android 10 or higher,
+     * PluginResult will be modified and returned in one of the callbacks
+     * of "triggerZebraSNRetrieval()"
+     *
+     * In any other cases or devices, PluginResult is returned immediately.
      *
      * @param action            The action to execute.
      * @param args              JSONArry of arguments for the plugin.
      * @param callbackContext   The callback id used when calling back into JavaScript.
      * @return                  True if the action was valid, false if not.
      */
     public boolean execute(String action, JSONArray args, CallbackContext callbackContext) throws JSONException {
+
+        this._callbackContext = callbackContext;
+
         if ("getDeviceInfo".equals(action)) {
-            JSONObject r = new JSONObject();
-            r.put("uuid", Device.uuid);
-            r.put("version", this.getOSVersion());
-            r.put("platform", this.getPlatform());
-            r.put("model", this.getModel());
-            r.put("manufacturer", this.getManufacturer());
-	        r.put("isVirtual", this.isVirtual());
-            r.put("serial", this.getSerialNumber());
-            callbackContext.success(r);
+
+            addToReturnValue("uuid", Device.uuid);
+            addToReturnValue("version", this.getOSVersion());
+            addToReturnValue("platform", this.getPlatform());
+            addToReturnValue("model", this.getModel());
+            addToReturnValue("manufacturer", this.getManufacturer());
+            addToReturnValue("isVirtual", this.isVirtual());
```

Comment:
To me it seems that there's no need for an additional function `addToReturnValue` (at least once `_callbackContext` and `_returnValue` are moved to parameters).